consensus/istanbul/core/types.go

```diff
@@ -134,10 +136,15 @@ func (m *message) FromPayload(b []byte, validateFn func([]byte, []byte) (common.
 			return err
 		}
 
-		_, err = validateFn(payload, m.Signature)
+		signerAdd, err := validateFn(payload, m.Signature)
+		if err != nil {
+			return err
+		}
+		if bytes.Compare(signerAdd.Bytes(), m.Address.Bytes()) != 0 {
```

Comment:
```suggestion
		if !bytes.Equal(signerAdd.Bytes(), m.Address.Bytes()) {
```
This might be better.